sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -27,11 +21,79 @@ AZ_NODISCARD az_result az_keyvault_keys_createKey(
     az_span keyname,
     az_span keytype,
     az_keyvault_keys_keys_options * options,
-    az_span * out) {
+    const az_mut_span * const out) {
   (void)client;
   (void)keyname;
   (void)keytype;
   (void)options;
   (void)out;
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_keyvault_build_url(
+    az_span const uri,
+    az_span const key_type,
+    az_span const key_name,
+    az_mut_span const out) {
+  az_span_builder s_builder = az_span_builder_create(out);
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, uri));
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, AZ_KEY_VAULT_URL_PATH_SEPARATOR));
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, key_type));
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, AZ_KEY_VAULT_URL_PATH_SEPARATOR));
+  AZ_RETURN_IF_FAILED(az_span_builder_append(&s_builder, key_name));
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_keyvault_keys_getKey(
+    az_keyvault_keys_client const * const client,
+    az_span const keyname,
+    az_key_vault_key_type const keytype,
+    const az_mut_span * const out) {
+  // create request buffer TODO: define size for a getKey Request
```

Comment:
I just want to say that I LOVE all the comments in this method - they're perfect!